blog/2023-07-13-reproducibility-designer.md

```diff
@@ -0,0 +1,41 @@
+---
+author: Antonin Delpeuch
+title: "Designer seeked for OpenRefine's reproducibility improvements"
+---
+
+OpenRefine is hiring a designer to help shape our improvements to OpenRefine's operation history and test those changes with user panels.
+<!--truncate-->
+
+[OpenRefine](https://openrefine.org/) is a powerful tool to clean messy data, popular in a diverse range of communities. It has been serving the needs of journalists, librarians, Wikipedians, scientists for more than ten years and is taught in many curricula and workshops around the world. OpenRefine is a fiscally sponsored project of [Code for Science & Society Inc](https://codeforscience.org/), a 501(c)(3) charitable organization in the USA.
+
+We are currently working on improving the reproducibility of OpenRefine workflows as part of [a project funded by the Essential Open Source Software for Science programme](https://chanzuckerberg.com/eoss/proposals/improving-openrefines-reproducibility/).
+This consists in improving the user experience around OpenRefine's operation history, which tracks all transformations applied on the original data. This work aims to address a range of long standing issues, such as:
+- limitations of the Undo/Redo functionality, such as the lack of ability to undo an operation in the middle of the history list ([#183](https://github.com/OpenRefine/OpenRefine/issues/183)) or unintentional loss of work after using the undo feature and
+  applying a new operation ([#369](https://github.com/OpenRefine/OpenRefine/issues/369), [#3184](https://github.com/OpenRefine/OpenRefine/issues/3184))
+- the difficulty in reusing and sharing OpenRefine workflows, due to the lack of error handling when applying series of operations and the difficulty to adapt a workflow to a new project (for instance, with different column names)
+- the lack of support for running OpenRefine workflows as part of larger pipelines, without using the web UI interactively. This covers for instance the lack of import metadata in the JSON export of the history
+  ([#460](https://github.com/OpenRefine/OpenRefine/issues/460)).
+
+See the corresponding [GitHub project](https://github.com/orgs/OpenRefine/projects/6) for more issues in the scope of this project.
+
+We are looking for a part-time designer who will
+* design changes to the user interface to support those improvements;
```

Comment:
One of the first things which needs to be done, in my opinion, is to design (or, preferably, choose) a DSL which meets the (implied) requirements above of being amenable to source control diffing & merging, easy(ier) for a human to edit, compatible with a parser which can generate high quality error messages for the user, etc.

Reply:
That makes sense, I'd be happy to list that explicitly. In my mind it is indeed part of the role but perhaps not the first task I would give them because it's feels more challenging than UI tweaks, especially for someone new-ish to the project.

Reply:
I've added a bullet point about this. I have not used the term "DSL" because I think it could be seen as too programmer-oriented, which could scare off good candidates (we're not expecting the designer to design a programming language).
But I agree that from a technical perspective this is an appropriate term to describe what we are trying to build.

Reply:
DSL wasn't intended for the copy. Since the domain is workflows, I'd probably have gone with "workflow language" rather than "representation of series of operations forming workflows." If you want to leave the door open for visual/graphical languages, perhaps just say that explicitly, "workflow language (textual or visual)."

I'll mark this as resolved and make a suggestion on the new verbiage.